src/components/topBars/MarkingToolbar/components/CandidateSelector/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { ChevronLeftIcon, ChevronRightIcon } from '@radix-ui/react-icons'
+import { Badge, Button, Flex } from '@radix-ui/themes'
+import { first, indexOf, isUndefined, last } from 'lodash'
+import React, { FC, ReactNode, useMemo } from 'react'
+import Select, { StylesConfig } from 'react-select'
+
+import { Student } from '../../../../../types/marking'
+import './index.css'
+
+type Option = {
+  value: string
+  label: string
+}
+
+type GroupedOption = {
+  label: string
+  options: Option[]
+}
+
+export interface CandidateSelectorProps {
+  students: Student[]
+  student: Student | undefined
+  markingStatus: { [_: string]: number }
+  onSelect: (student: Student | undefined) => void
+}
+
+const CandidateSelector: FC<CandidateSelectorProps> = ({
+  students,
+  student,
+  markingStatus,
+  onSelect,
+}) => {
+  const groupedOptions: GroupedOption[] = useMemo(
+    () => [
+      {
+        label: 'Students',
+        options: students.map((s) => ({ value: s.username, label: s.cid })),
+      },
+    ],
+    [students]
+  )
+  const value = useMemo(
+    () =>
+      isUndefined(student)
+        ? undefined
+        : {
+            value: student.username,
+            label: student.cid,
+          },
+    [student]
+  )
+
+  function handleChange(newValue) {
+    onSelect(students.find((s) => s.username === newValue.value))
+  }
+
+  const prev = () => onSelect(students[indexOf(students, student) - 1])
+  const next = () => onSelect(students[indexOf(students, student) + 1])
+
+  const formatGroupLabel = (data): ReactNode => (
+    <Flex justify="between">
+      <span>{data.label}</span>
+      <Badge color="gray" radius="full">
+        {data.options.length}
+      </Badge>
+    </Flex>
+  )
+
+  const formatOptionLabel = (data): ReactNode => {
+    const sectionsRemaining = markingStatus[data.value]
+    return (
+      <Flex justify="between">
+        <span>{data.label}</span>
+        <Badge color={!sectionsRemaining ? 'green' : 'red'} variant="solid" radius="full">
```

Comment:
Hi
I think you need to account for marking not started. When no marks are entered then you see an empty (no number) green pill. Im' just trying to work out the markingStatus for no marks entered.

Reply:
Ah, ok markingStatus[data.value] is undefined where no marks have been recorded/entered.